src/bosh-director/lib/bosh/director/orphan_network_manager.rb

```diff
@@ -0,0 +1,91 @@
+module Bosh::Director
+  class OrphanNetworkManager
+    def initialize(logger)
+      @logger = logger
+      @transactor = Transactor.new
+    end
+
+    def orphan_network(network)
+      @transactor.retryable_transaction(Bosh::Director::Config.db) do
+        begin
+          parent_id = add_event('orphan', network.name, 'network')
+          network.orphaned = true
+          network.orphaned_at = Time.now
+          @logger.info("Orphaning network: '#{network.name}'")
+          network.save
+        rescue StandardError => e
+          raise e
+        ensure
+          add_event(
+            'orphan',
+            network.name,
+            'network',
+            parent_id,
+            e,
+          )
+        end
+      end
+    end
+
+    def unorphan_network(network)
+      @transactor.retryable_transaction(Bosh::Director::Config.db) do
+        network.orphaned = false
+        network.save
+      end
+    end
+
+    # returns a list of orphaned networks
+    def list_orphan_networks
+      Models::Network.where(orphaned: true).map do |network|
+        {
+          'name' => network.name,
+          'type' => network.type,
+          'created_at' => network.created_at.to_s,
+          'orphaned_at' => network.orphaned_at.to_s,
+        }
+      end
+    end
+
+    def delete_network(network)
```

Comment:
which component will be calling delete_network?It might be a good idea to add network lock before deleting the network to avoid race condition where the same network is being reused by create_networks function

Reply:
@GarfieldIsAPhilosopher For this PR, no components are calling delete network. Ina future PR, it will be called. Good point about locking!